### PR TITLE
Remove force unwrapping of incoming url parameters

### PR DESCRIPTION
**Pull request by MarkPerkins**
## What does this PR do?

Avoid unnecessary crashes due to unrelated incoming URLs by:
- guard against existence of URL components.
- guard against query parameters
- validate required cookies *bonus: discard result from user defaults synchronization.

## Test Plan

Didn't see any unit test files, otherwise i'd add code coverage for them. Happy to do it if there is a testing scheme!

- Create sample project with appwrite sdk
- establish universal links
- apply `.registerOAuthHandler()` to root view.
- start firing deeplinks, which will crash on open.

## Related PRs and Issues
Discord
https://discord.com/channels/564160730845151244/1252726652044050572

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Most of it.  Followed PR naming, and added a issue on github.

### CODE REVIEW

File: templates/swift/Sources/OAuth/WebAuthComponent.swift.twig [hunk 1]
```diff
@@ -103,6 +111,15 @@ public class WebAuthComponent {
...
 
+    private static func validateRequiredCookieParts(_ cookieParts: [String: String]) -> (domain: String, key: String, secret: String)? {
+        guard let domain = cookieParts["domain"],
+                let key = cookieParts["key"],
+                let secret = cookieParts["secret"] else {
+            return nil
+        }
+        return (domain, key, secret)
+    }
+
```
Comment (abnegate) [inline templates/swift/Sources/OAuth/WebAuthComponent.swift.twig:122]: Let's do this inline instead of a separate function

Comment (MarkPerkins) [reply templates/swift/Sources/OAuth/WebAuthComponent.swift.twig:122]: sound good.

File: templates/swift/Sources/OAuth/WebAuthComponent.swift.twig [hunk 2]
```diff
@@ -53,17 +53,28 @@ public class WebAuthComponent {
...
+              let key = cookieParts["key"],
+              let secret = cookieParts["secret"] else {
+            return
+        }
+    
+        domain.remove(at: domain.startIndex)
+        var domain = validatedCookieParts.domain
         domain.remove(at: domain.startIndex)
+        let key = validatedCookieParts.key
+        let secret = validatedCookieParts.secret
```
Comment (abnegate) [inline templates/swift/Sources/OAuth/WebAuthComponent.swift.twig:76]: We can remove these lines now

File: templates/swift/Sources/OAuth/WebAuthComponent.swift.twig [hunk 3]
```diff
@@ -92,7 +103,7 @@ public class WebAuthComponent {
             cookie += "; secure"
         }
 
-        let existing = UserDefaults.standard.stringArray(forKey: domain)
+        _ = UserDefaults.standard.stringArray(forKey: domain)
```
Comment (abnegate) [inline templates/swift/Sources/OAuth/WebAuthComponent.swift.twig:106]: Not sure we even need this at all if the read result can be discarded 🤔

File: templates/swift/Sources/OAuth/WebAuthComponent.swift.twig (hunk 2 above)
Comment (MarkPerkins) [reply templates/swift/Sources/OAuth/WebAuthComponent.swift.twig:76]: ha.  a reminder to not use the github editor.

File: templates/swift/Sources/OAuth/WebAuthComponent.swift.twig (hunk 3 above)
Comment (MarkPerkins) [reply templates/swift/Sources/OAuth/WebAuthComponent.swift.twig:106]: I was also not aware of the purpose, this just silenced the obvious compiler warning.  I wonder if it was part of a previous attempt to set the cookie for the domain.

Comment (abnegate) [reply templates/swift/Sources/OAuth/WebAuthComponent.swift.twig]: Yep probably my own leftover 😅

**Review by abnegate — APPROVED** [review]
Nice work 💪

**Pull request merged**